src/coreComponents/mesh/generators/MeshGeneratorBase.cpp

```diff
@@ -47,15 +48,26 @@ MeshGeneratorBase::CatalogInterface::CatalogType & MeshGeneratorBase::getCatalog
   return catalog;
 }
 
-CellBlockManagerABC & MeshGeneratorBase::generateMesh( Group & parent, array1d< int > const & partition )
+void MeshGeneratorBase::generateMesh( Group & parent, SpatialPartition & partition )
 {
-  CellBlockManager & cellBlockManager = parent.registerGroup< CellBlockManager >( keys::cellManager );
+  MeshBody & meshBody = dynamic_cast< MeshBody & >( parent );
+  if( meshBody.hasParticles() )
+  {
+    ParticleBlockManager & particleBlockManager = parent.registerGroup< ParticleBlockManager >( keys::particleManager );
 
-  fillCellBlockManager( cellBlockManager, partition );
+    MeshLevel & meshLevel0 = meshBody.getBaseDiscretization();
+    ParticleManager & particleManager = meshLevel0.getParticleManager();
 
-  this->attachWellInfo( cellBlockManager );
+    fillParticleBlockManager( particleBlockManager, particleManager, partition );
+  }
+  else
+  {
+    CellBlockManager & cellBlockManager = parent.registerGroup< CellBlockManager >( keys::cellManager );
 
-  return cellBlockManager;
+    fillCellBlockManager( cellBlockManager, partition );
+
+    this->attachWellInfo( cellBlockManager );
+  }
```

Comment:
This part of the code should not we aware of neither `MeshBody` nor `MeshLevel`, as they basically depend on the whole `geos`.
It took us lots of efforts (@XL64 and myself) to slowly but surely remove these dependencies from the geometry world. Is it possible to keep this in the `MeshManager` part?